tooling/templatize/pkg/ev2/mapping_test.go

```diff
@@ -0,0 +1,34 @@
+package ev2
+
+import (
+	"fmt"
+	"testing"
+
+	"gotest.tools/assert"
+
+	"github.com/Azure/ARO-HCP/tooling/templatize/pkg/config"
+)
+
+func TestMapping(t *testing.T) {
+	testData := config.Variables{
+		"key1": "value1",
+		"key2": 42,
+		"key3": true,
+		"parent": map[string]interface{}{
+			"nested": "nestedvalue",
+			"deeper": map[string]interface{}{
+				"deepest": "deepestvalue",
+			},
+		},
+	}
+	expectedFlattened := map[string]string{
+		"__KEY1__":                  "key1",
+		"__KEY2__":                  "key2",
+		"__KEY3__":                  "key3",
+		"__PARENT_NESTED__":         "parent.nested",
+		"__PARENT_DEEPER_DEEPEST__": "parent.deeper.deepest",
+	}
+	flattened, replace := EV2Mapping(testData, []string{})
+	assert.DeepEqual(t, expectedFlattened, flattened)
```

Comment:
I'd suggest `assert.Empty(t, cmp.Diff(expectedFlattened, flattened))` - the outputs on failure are much nicer.

Reply:
nice. adopted